catroid/src/main/java/org/catrobat/catroid/ui/ExportLaunchers.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2019 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.ui
+
+import android.content.Intent
+import android.support.v4.content.FileProvider
+import org.catrobat.catroid.stage.StageActivity
+import java.io.File
+
+interface ExportLauncher {
+
+    fun startActivityForResult(requestCode: Int)
+}
+
+class ExportEmbroideryFileLauncher(
+    private val activity: StageActivity,
+    private val file: File
+) : ExportLauncher {
+
+    override fun startActivityForResult(requestCode: Int) {
+        val fileUri = FileProvider.getUriForFile(activity, activity.packageName + ".fileProvider", file)
+        val shareIntent = Intent(Intent.ACTION_SEND)
+        shareIntent.type = "text/plain"
```

Comment:
this mimetype "text/plain" leads to all apps being listed that can deal with plain text.
for me that is:
all messenger apps,
copy to clipboard (!!)
calenmdar,
sms,
bluetooth,
etc...

check if there is a more fitting mimetype, eg. making it csv will then only list messenger apps and apps that can deal with files to be uploaded/send somewhere.

Reply:
i just looked it up:
application/octet-stream

would be the correct mimetype for this kind of stuff

Reply:
facebook messenger also wouldnt allow me to send the file if it was text/plain and just doesnt do anything, and for this mimetype now only apps that can really deal with it are offered.